test/orbit_determination.jl

```diff
@@ -0,0 +1,143 @@
+# This file is part of the NEOs.jl package; MIT licensed
+
+using NEOs
+using PlanetaryEphemeris
+using LinearAlgebra
+using Test
+
+using NEOs: NEOSolution, numberofdays
+
+@testset "Orbit Determination" begin
+    @testset "Gauss Method" begin
+        # Load observations 
+        radec = read_radec_mpc(joinpath("data", "RADEC_2023_DW.dat"))
+        # Parameters
+        params = Parameters(abstol = 1e-20, order = 25, parse_eqs = true)
+    
+        # Orbit Determination
+        sol = orbitdetermination(radec, params)
+    
+        # Values by December 21, 2023
+        
+        # Vector of observations
+        @test length(radec) == 123
+        @test numberofdays(radec) < 21.0
+        # Orbit solution
+        @test isa(sol, NEOSolution{Float64, Float64})
+        # Observation nights
+        @test length(sol.nights) == 44
+        @test sol.nights[1].radec[1] == radec[1]
+        @test sol.nights[end].radec[end] == radec[end]
+        @test issorted(sol.nights)
+        # Backward integration
+        @test datetime2days(date(radec[1])) > sol.bwd.t0 + sol.bwd.t[end]
+        @test all( norm.(sol.bwd.x, Inf) .< 2 )
+        @test isempty(sol.t_bwd) && isempty(sol.x_bwd) && isempty(sol.g_bwd)
+        # Forward integration
+        @test datetime2days(date(radec[end])) < sol.fwd.t0 + sol.fwd.t[end]
+        @test all( norm.(sol.fwd.x, Inf) .< 2 )
+        @test isempty(sol.t_fwd) && isempty(sol.x_fwd) && isempty(sol.g_fwd)
+        # Vector of residuals
+        @test length(sol.res) == 123
+        @test iszero(count(outlier.(sol.res)))
+        # Least squares fit
+        @test sol.fit.success
+        @test all( sigmas(sol) .< 1e-4 )
+        @test nrms(sol) < 0.38
+        # Scalig factors
+        @test all(sol.scalings .== 1e-6)
+    end
+
+    @testset "Too Short Arc" begin

```

Comment:
Thanks for adding these tests! Have you tried these subroutines with objects such as 2008TC3 and 2014AA? I think those would be ideal test cases...

Reply:
Thank you for the suggestion, both are indeed very interesting asteroids. Currently,  2014 AA converges to an orbit without a problem via `orbitdetermination`. On the other hand, the routine is not well suited for an object like 2008 TC3 that has a lot of observations covering a relatively short time. In future PRs I will address performance and converge issues like this.

Reply:
> Currently, 2014 AA converges to an orbit without a problem via orbitdetermination. 

Nice, then I suggest to add that 2014 AA to the tests!

> On the other hand, the routine is very slow for an object like 2008 TC3 that has > 800 observations spanning < 1 day

What happens if you try doing the same but using a tracklet with only the first four observations from Mt. Lemmon in the MPC astrometry file for 2008 TC3?

Reply:
> Nice, then I suggest to add that 2014 AA to the tests!

2014 AA is now part of the tests.

> What happens if you try doing the same but using a tracklet with only the first four observations from Mt. Lemmon in the MPC astrometry file for 2008 TC3?

It is posible to get an orbit using `orbitdetermination` with the 15 observations before 6/Oct/08 10:00 UTC and then add the remaining observations via `propres` + `tryls`. I've also added this handcrafted OD solution to the tests.

Reply:
Awesome, thanks for adding these tests!